src/features/vsProvider.ts

```diff
@@ -152,21 +218,48 @@ export default class ValeServerProvider implements vscode.CodeActionProvider {
           });
       } else {
         // We're using the CLI ...
-        //
-        // TODO
+        try {
+          const result = await this.runVale(textDocument);
+      } catch (error) {
+        // TODO: Error handling
+      }
+
       }
   }
 
+  private async runVale (file: vscode.TextDocument) {
+    const binaryLocation = readBinaryLocation();
+    const configLocation = readFileLocation()!;
+    // TODO: Hmm
+    this.stylesPath = "/Users/chrisward/Workspace/TesttheDocs";
```

Comment:
This in particular needs thought. I am not sure how to handle the equivalent in the CLI version, so it's hard coded right now while I was testing. To make this work properly, the path needs to follow the usual Vale CLi scoping rules.

Reply:
You can get the path by inspecting the output of `ls-config` once you have the `binaryLocation` and `configLocation`:

```console
vale --config="/Users/jdkato/Desktop/docs/.vale.ini" ls-config
```